WordPress/Classes/Services/PlanService.swift

```diff
@@ -223,44 +223,33 @@ extension PlanService {
                         success()
                         return
                 }
-                PlanStorage.updateHasDomainCredit(planIdInt,
-                                                  forSite: siteID,
-                                                  hasDomainCredit: plans.activePlan.hasDomainCredit ?? false)
-                success()
-        },
+                ContextManager.shared.performAndSave({ context in
+                    PlanStorage.updateHasDomainCredit(
+                        planIdInt,
+                        forSite: siteID,
+                        hasDomainCredit: plans.activePlan.hasDomainCredit ?? false,
+                        in: context
+                    )
+                }, completion: success, on: .main)
```

Comment:
Any plan to make the queue injectable in the future?

I see this is an `@objc` method, so there might be additional friction. Also I understand it might be safe to assume `.main` to be the appropriate queue for this. 

Still, maybe I have the wrong mental model, but it doesn't feel right to me for a service object or anything else that is not immediately connected to the view layer to _explicitly_ schedule work on the main thread.